### PR TITLE
neon: try to fix broken main build

### PR DESCRIPTION
**Pull request by rawlingsj**
```
linking with cc failed: exit status: 1
undefined reference to '__pthread_cond_timedwait64'
```

### CODE REVIEW

**Review by imjasonh — APPROVED** [review]

**Comment by rawlingsj** [comment]
```
2024/07/26 12:30:17 WARN   = note: /usr/lib/gcc/x86_64-pc-linux-gnu/13.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /home/build/target/release/deps/libtikv_jemalloc_sys-f576629b54940529.rlib(background_thread.pic.o): in function `background_thread_sleep':
2024/07/26 12:30:17 WARN           /home/build/target/release/build/tikv-jemalloc-sys-90af8d64ec9f00fb/out/build/src/background_thread.c:159:(.text.background_thread_sleep.isra.0+0xe5): undefined reference to `__pthread_cond_timedwait64'
2024/07/26 12:30:17 WARN           collect2: error: ld returned 1 exit status
2024/07/26 12:30:17 WARN           
2024/07/26 12:30:17 WARN   = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
2024/07/26 12:30:17 WARN   = note: use the `-l` flag to specify native libraries to link
2024/07/26 12:30:17 WARN   = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)
2024/07/26 12:30:17 WARN
2024/07/26 12:30:17 WARN error: could not compile `proxy` (bin "proxy") due to 1 previous error
```

maybe adding `libpthread-stubs` wasn't the right thing to try, going to revert the neon update that is breaking main https://github.com/wolfi-dev/os/pull/24973

Might need a hand with this one.

**Comment by rawlingsj** [comment]
The revert has failed too, so something else must have gone into wolfi

File: neon.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ environment:
       - icu-dev
       - libcurl-openssl4
       - libpq-16
+      - libpthread-stubs
```
Comment (xnox) [inline neon.yaml:29]: This is an empty package which does nothing.

**Comment by ajayk** [comment]
Closing this as this is resolved now

**Pull request closed without merging**